### PR TITLE
Adapt to changes from code_coverage

### PR DESCRIPTION
**Pull request by martiniil**
Due to recent changes in mikeferguson/code_coverage#15 I spent some effort in making the coverage target work as expected.
```
catkin_make -DENABLE_COVERAGE_TESTING=ON -DCMAKE_BUILD_TYPE=Debug PACKAGE_NAME_coverage
```
In the end I reviewed all test-target definitions, since many dependencies were missing or not needed.

### CODE REVIEW

File: prbt_hardware_support/test/unittests/unittest_modbus_api_spec.cpp [hunk 1]
```diff
@@ -111,6 +111,5 @@ TEST(ModbusApiSpecTest, NodeHandleConstructionMissingApiSpec)
 
 int main(int argc, char **argv){
   testing::InitGoogleTest(&argc, argv);
-  ros::init(argc, argv, "unittest_modbus_api_spec");
```
Comment (agutenkunst) [inline prbt_hardware_support/test/unittests/unittest_modbus_api_spec.cpp]: Is this related to the intention of this PR?

**Comment by agutenkunst** [comment]
Just one comment. Otherwise thank you for you contribution!
Are you up for a backport to kinetic? Should we Release?

File: prbt_hardware_support/test/unittests/unittest_modbus_api_spec.cpp (hunk 1 above)
Comment (martiniil) [reply prbt_hardware_support/test/unittests/unittest_modbus_api_spec.cpp]: No

**Comment by martiniil** [comment]
Since b105f67 it doesn't work on my local PC. Maybe related to mikeferguson/code_coverage#14 ?

**Comment by martiniil** [comment]
Since it is again unclear to me, how we deal with this situation (`pilz_utils` coverage job fails) I swapped out the other adjustements in CMakeLists to #290.

**Comment by martiniil** [comment]
Applied following changes to kinetic: d6cf843

**Comment by jschleicher** [comment]
@martiniil Is this still relevant? There were some PRs for coverage merged recently. Please update conflicts or close here.
